# (PART\*) DIFFERENTIAL EXPRESSION {-}

# Transforming Data

The following excerpt comes from @Ritchie2015 :

> _Measuring expression in multiple RNA samples produces columns of correlated expression values, which are highly correlated because they are measured on the same set of genes or genomic features. It has long been established in the biomedical literature that the level of agreement between correlated variables can be usefully examined by plotting differences versus means._

In other words, gene expression data is full of correlations. We must think carefully about how we examine and plot gene expression data. In the next steps, we will:

- Examine how $\log_{2}$ transformations and fold-change improve data clarity
- Learn how to make MA plots on gene expression data

Much of the following has been adopted from the [`Glimma` vignette](https://bioconductor.org/packages/release/bioc/vignettes/Glimma/inst/doc/limma_edger.html) for `limma` and `edgeR` [@Su2017].

## Gene Expression Data

First, we will load the necessary packages.

```{r, warning = FALSE, message = FALSE, echo = FALSE}
# Needed for GHA
install.packages("https://cran.r-project.org/src/contrib/matrixStats_0.61.0.tar.gz", repo=NULL, type="source")
BiocManager::install(c("MatrixGenerics", "DelayedArray", "SummarizedExperiment"), update = FALSE, ask = FALSE)
BiocManager::install(c("Glimma", "limma"), update = FALSE, ask = FALSE)
```

```{r, warning = FALSE, message = FALSE}
# Install and load Glimma, limma, and matrixStats
# AnVIL::install(c("Glimma", "limma"))
library(Glimma)
library(limma)
library(matrixStats)
```

Load the gene expression data. We will be using data from @Sheridan2015 that has already been vetted to demo plot creation. This experiment consists of three cell populations - basal, luminal progenitor (LP) and mature luminal (ML) - sorted from mice mammary glands. Each treatment consists of three replicates. The gene expression data is stored in a special list-based data class called a "DGEList" (this stands for Digital Gene Expression).

```{r}
# Load the gene expression data
dge <- readRDS(system.file("RNAseq123/dge.rds", package = "Glimma"))
head(dge$counts,10)
```

In this data, each sample is in a column, while each gene is in a row. These raw counts range from very small to quite large numbers!

# Plotting Expression Data 

## Plotting Raw Counts

Let's say we want to compare basal and LP samples. We will first look at the metadata.

```{r}
# View metadata
dge$samples
```

We will manually take the mean across basal samples and LP samples.

```{r}
# Collect the counts
raw_counts <- dge$counts
# Basal samples are columns 3, 4, and 7
basal_sample_means <- rowMeans2(raw_counts[, c(3, 4, 7)])
# LP samples are columns 1, 6, and 9
LP_sample_means <- rowMeans2(raw_counts[, c(1, 6, 9)])
```

Then we will plot the two groups.

```{r}
plot(basal_sample_means, LP_sample_means)
```

This is hard to interpret. Most of the genes are clustered in the bottom left corner.

## Transforming ($\log_{2}$)

Using a $\log_{2}$ transformation makes it easier to examine all the genes together.

```{r}
# Plot with log2 transformation
plot(log2(basal_sample_means), log2(LP_sample_means))
```

## Using Fold-Change

As stated at the very start of this chapter, plotting differences versus means can be very helpful when many genes are correlated. It also makes interpretation easier when combined with a $\log_{2}$ transformation.

When basal and LP expression are equal, fold-change is equal to zero.

```{r}
log2(10) - log2(10)
```

When basal expression is greater, fold-change is positive.

```{r}
log2(20) - log2(10)
log2(100) - log2(1)
```

However when LP expression is greater, fold-change is negative.

```{r}
log2(10) - log2(20)
log2(1) - log2(100)
```

Calculate fold-change for the data.

```{r}
log2_fold_change <- log2(basal_sample_means) - log2(LP_sample_means)
mean_expression <- (log2(basal_sample_means) + log2(LP_sample_means)) / 2
```

Plot the values above, with mean expression on the x-axis and fold-change on the y-axis.

```{r}
plot(mean_expression, log2_fold_change)
```

This gives us a very rough idea of how transformation and using fold-change can aid in interpretation of the data. In reality, we need to cover a few more steps before creating this kind of plot.

# Creating MA Plots

In the previous steps, you created a preliminary MA plot. MA plots are a widely used way to visualize genomic data. As you experienced above, these plots are a specific type of scatterplot typically used for gene expression datasets. The _**M**_ represents the difference between two conditions (fold-change), while the _**A**_ represents the average intensity of the expression. Both values take on a $\log_{2}$ transformation.

_**M**_ is expressed as a log ratio or difference in the following form. _**M**_ is almost always placed on the y-axis.

$$M = log_{2}(\frac{condition 1}{condition 2}) = log_{2}(condition 1) - log_{2}(condition2)$$

_**A**_ is more simple, taking the form of a transformed average. _**A**_ is often called "log mean expression" and is almost always placed on the x-axis.

$$A = \frac{1}{2} (log_{2}(condition 1) + log_{2}(condition 2))$$
In the next steps, we will go over how to create an MA plot using `limma`. First, we will need to supply the design matrix and contrast matrix.

## The Design Matrix

Load the design matrix. This indicates which samples correspond to which cell populations.

```{r}
# Load the experimental design matrix
design <- readRDS(
  system.file("RNAseq123/design.rds", package = "Glimma"))
design
```

## The Contrast Matrix

Load the contrast matrix. This indicates which cell populations we want to compare.

```{r}
contr.matrix <- readRDS(
  system.file("RNAseq123/contr.matrix.rds", package = "Glimma"))
```

## Fitting the Data

There are several more steps to normalize and fit the data before making an MA plot. Don't worry! We will cover these steps in more detail later in the course. Briefly, the `voom()` function $\log_{2}$ transforms and normalizes the data. `lmFit()` fits a linear model for each gene to examine differential expression between the cell populations we want to compare.

```{r}
v <- voom(dge, design)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts = contr.matrix)
```

## Creating an MA Plot

A simple MA plot can be produced with the `plotMA()` function. Other packages for exploring differential gene expression in R, such as `DESeq2`, also have functions to create MA plots.

```{r}
plotMA(vfit)
```

## Interactive MA Plots

`glimmaMA()` from the `Glimma` package creates interactive MA plots. Again, don't worry about the preparation steps - these will be covered in more detail later. `eBayes()` computes statistics by empirical Bayes moderation of the standard errors towards a global value.

```{r}
efit <- eBayes(vfit)
glimmaMA(efit, dge = dge)
```

## Recap

```{r}
sessionInfo()
```
