# (PART\*) DIFFERENTIAL EXPRESSION {-}

# Transforming Data

The following excerpt comes from @Ritchie2015 :

> _Measuring expression in multiple RNA samples produces columns of correlated expression values, which are highly correlated because they are measured on the same set of genes or genomic features. It has long been established in the biomedical literature that the level of agreement between correlated variables can be usefully examined by plotting differences versus means._

In other words, gene expression data is full of correlations. We must think carefully about how we examine and plot gene expression data. In the next steps, we will:

- Examine how $\log_{2}$ transformations and fold-change improve data clarity
- Learn how to make MA plots on gene expression data

Much of the following has been adopted from the [`Glimma` vignette](https://bioconductor.org/packages/release/bioc/vignettes/Glimma/inst/doc/limma_edger.html) for `limma` and `edgeR` [@Su2017].

## Gene Expression Data

First, we will load the necessary packages.

```{r, warning = FALSE, message = FALSE, echo = FALSE}
# Needed for GHA
install.packages("https://cran.r-project.org/src/contrib/matrixStats_0.61.0.tar.gz", repo=NULL, type="source")
BiocManager::install(c("MatrixGenerics", "DelayedArray", "SummarizedExperiment"), update = FALSE, ask = FALSE)
BiocManager::install(c("Glimma", "limma"), update = FALSE, ask = FALSE)
```

```{r, warning = FALSE, message = FALSE}
# Install and load Glimma, limma, and matrixStats
# AnVIL::install(c("Glimma", "limma"))
library(Glimma)
library(limma)
library(matrixStats)
```

Load the gene expression data. We will be using data from @Sheridan2015 that has already been vetted to demo plot creation. This experiment consists of three cell populations - basal, luminal progenitor (LP) and mature luminal (ML) - sorted from mice mammary glands. Each treatment consists of three replicates. The gene expression data is stored in a special list-based data class called a "DGEList" (this stands for Digital Gene Expression).

```{r}
# Load the gene expression data
dge <- readRDS(system.file("RNAseq123/dge.rds", package = "Glimma"))
head(dge$counts,10)
```

In this data, each sample is in a column, while each gene is in a row. These raw counts range from very small to quite large numbers!
