<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Simple Model Fitting | Statistics for Genomics: Differential Expression</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Simple Model Fitting | Statistics for Genomics: Differential Expression" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Simple Model Fitting | Statistics for Genomics: Differential Expression" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2023-03-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/GDSCN_style/gdscn_favicon.ico" type="image/x-icon" />
<link rel="prev" href="design-matrices.html"/>
<link rel="next" href="help.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
<link rel="stylesheet" href="assets/GDSCN_style/gdscn.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://www.gdscn.org/" target="_blank"><img src="assets/GDSCN_style/logo-gdscn.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-collection"><i class="fa fa-check"></i>AnVIL Collection</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html"><i class="fa fa-check"></i><b>1</b> Overview and Learning Objectives</a>
<ul>
<li class="chapter" data-level="1.1" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html#activity-context"><i class="fa fa-check"></i><b>1.1</b> Activity Context</a></li>
<li class="chapter" data-level="1.2" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html#learning-objectives"><i class="fa fa-check"></i><b>1.2</b> Learning Objectives</a></li>
</ul></li>
<li class="part"><span><b>RStudio on AnVIL</b></span></li>
<li class="chapter" data-level="2" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>2</b> Getting Set Up</a>
<ul>
<li class="chapter" data-level="2.1" data-path="getting-set-up.html"><a href="getting-set-up.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="getting-set-up.html"><a href="getting-set-up.html#create-a-google-account"><i class="fa fa-check"></i><b>2.2</b> Create a Google Account</a></li>
<li class="chapter" data-level="2.3" data-path="getting-set-up.html"><a href="getting-set-up.html#sign-in-to-terra"><i class="fa fa-check"></i><b>2.3</b> Sign in to Terra</a></li>
<li class="chapter" data-level="2.4" data-path="getting-set-up.html"><a href="getting-set-up.html#share-your-id"><i class="fa fa-check"></i><b>2.4</b> Share your ID</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="workspace.html"><a href="workspace.html"><i class="fa fa-check"></i><b>3</b> Creating a Workspace</a></li>
<li class="chapter" data-level="4" data-path="rstudio.html"><a href="rstudio.html"><i class="fa fa-check"></i><b>4</b> Working in RStudio</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rstudio.html"><a href="rstudio.html#launch-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.1</b> Launch RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.2" data-path="rstudio.html"><a href="rstudio.html#tour-rstudio"><i class="fa fa-check"></i><b>4.2</b> Tour RStudio</a></li>
<li class="chapter" data-level="4.3" data-path="rstudio.html"><a href="rstudio.html#more-practice-with-isee"><i class="fa fa-check"></i><b>4.3</b> More Practice with <code>iSEE</code></a></li>
<li class="chapter" data-level="4.4" data-path="rstudio.html"><a href="rstudio.html#stopping"><i class="fa fa-check"></i><b>4.4</b> Pause RStudio</a></li>
<li class="chapter" data-level="4.5" data-path="rstudio.html"><a href="rstudio.html#delete-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.5</b> Delete RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.6" data-path="rstudio.html"><a href="rstudio.html#video-guide"><i class="fa fa-check"></i><b>4.6</b> Video Guide</a></li>
</ul></li>
<li class="part"><span><b>EXPRESSION DATA</b></span></li>
<li class="chapter" data-level="5" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html"><i class="fa fa-check"></i><b>5</b> The <code>SummarizedExperiment</code> class</a>
<ul>
<li class="chapter" data-level="5.1" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html#exploring-summarizedexperiment"><i class="fa fa-check"></i><b>5.2</b> Exploring <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="5.3" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html#subsetting-summarizedexperiment"><i class="fa fa-check"></i><b>5.3</b> Subsetting <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="5.4" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html#recap"><i class="fa fa-check"></i><b>5.4</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html"><i class="fa fa-check"></i><b>6</b> Transformation, Fold-change, &amp; MA Plots</a>
<ul>
<li class="chapter" data-level="6.1" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#gene-expression-data"><i class="fa fa-check"></i><b>6.1</b> Gene Expression Data</a></li>
<li class="chapter" data-level="6.2" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#plotting-raw-counts"><i class="fa fa-check"></i><b>6.2</b> Plotting Raw Counts</a></li>
<li class="chapter" data-level="6.3" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#transforming-log_2"><i class="fa fa-check"></i><b>6.3</b> Transforming (<span class="math inline">\(\log_{2}\)</span>)</a></li>
<li class="chapter" data-level="6.4" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#using-fold-change-to-create-an-ma-plot"><i class="fa fa-check"></i><b>6.4</b> Using Fold-Change to Create an MA Plot</a></li>
<li class="chapter" data-level="6.5" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#future-directions"><i class="fa fa-check"></i><b>6.5</b> Future Directions</a></li>
<li class="chapter" data-level="6.6" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#recap-1"><i class="fa fa-check"></i><b>6.6</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="design-matrices.html"><a href="design-matrices.html"><i class="fa fa-check"></i><b>7</b> Design Matrices</a>
<ul>
<li class="chapter" data-level="7.1" data-path="design-matrices.html"><a href="design-matrices.html#overview-1"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="design-matrices.html"><a href="design-matrices.html#types-of-models"><i class="fa fa-check"></i><b>7.2</b> Types of Models</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="design-matrices.html"><a href="design-matrices.html#regression-continuous-variables"><i class="fa fa-check"></i><b>7.2.1</b> Regression: Continuous Variables</a></li>
<li class="chapter" data-level="7.2.2" data-path="design-matrices.html"><a href="design-matrices.html#means-categorical-variables"><i class="fa fa-check"></i><b>7.2.2</b> Means: Categorical Variables</a></li>
<li class="chapter" data-level="7.2.3" data-path="design-matrices.html"><a href="design-matrices.html#mean-reference-categorical-variables"><i class="fa fa-check"></i><b>7.2.3</b> Mean-Reference: Categorical Variables</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="design-matrices.html"><a href="design-matrices.html#making-a-design-matrix"><i class="fa fa-check"></i><b>7.3</b> Making a Design Matrix</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="design-matrices.html"><a href="design-matrices.html#regression-continuous-variables-1"><i class="fa fa-check"></i><b>7.3.1</b> Regression: Continuous Variables</a></li>
<li class="chapter" data-level="7.3.2" data-path="design-matrices.html"><a href="design-matrices.html#means"><i class="fa fa-check"></i><b>7.3.2</b> Means: Categorical Variables</a></li>
<li class="chapter" data-level="7.3.3" data-path="design-matrices.html"><a href="design-matrices.html#meanref"><i class="fa fa-check"></i><b>7.3.3</b> Mean-Reference: Categorical Variables</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="design-matrices.html"><a href="design-matrices.html#more-complex-designs"><i class="fa fa-check"></i><b>7.4</b> More Complex Designs</a></li>
<li class="chapter" data-level="7.5" data-path="design-matrices.html"><a href="design-matrices.html#inpractice"><i class="fa fa-check"></i><b>7.5</b> Design Matrix in Practice</a></li>
<li class="chapter" data-level="7.6" data-path="design-matrices.html"><a href="design-matrices.html#recap-2"><i class="fa fa-check"></i><b>7.6</b> Recap</a></li>
</ul></li>
<li class="part"><span><b>PUTTING IT TOGETHER</b></span></li>
<li class="chapter" data-level="8" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html"><i class="fa fa-check"></i><b>8</b> Simple Model Fitting</a>
<ul>
<li class="chapter" data-level="8.1" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#overview-2"><i class="fa fa-check"></i><b>8.1</b> Overview</a></li>
<li class="chapter" data-level="8.2" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#load-prepare-data"><i class="fa fa-check"></i><b>8.2</b> Load &amp; Prepare Data</a></li>
<li class="chapter" data-level="8.3" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#creating-the-design-matrix"><i class="fa fa-check"></i><b>8.3</b> Creating the Design Matrix</a></li>
<li class="chapter" data-level="8.4" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#testing-with-lmfit-and-ebayes"><i class="fa fa-check"></i><b>8.4</b> Testing with <code>lmFit</code> and <code>eBayes</code></a></li>
<li class="chapter" data-level="8.5" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#interpreting-toptable-output"><i class="fa fa-check"></i><b>8.5</b> Interpreting <code>topTable()</code> Output</a></li>
<li class="chapter" data-level="8.6" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#recap-3"><i class="fa fa-check"></i><b>8.6</b> Recap</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="help.html"><a href="help.html"><i class="fa fa-check"></i><b>A</b> Help</a></li>
<li class="chapter" data-level="B" data-path="give-us-feedback.html"><a href="give-us-feedback.html"><i class="fa fa-check"></i><b>B</b> Give Us Feedback</a></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
<p style="padding-left: 40px;"><div class="trapezoid" style = "padding-left: 40px;"><span>  <a href="https://forms.gle/AK12iVXTjsB7yCUUA"> Click here to provide feedback</a> <img src="assets/itcr_arrow.png" style=" width: 10%" ></span></div></p>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistics for Genomics: Differential Expression</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
</head>



<div class="hero-image-container">
  <img class= "hero-image" src="assets/GDSCN_style/dasl_thin_main_image.png">
</div>
<div id="simple-model-fitting" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> Simple Model Fitting</h1>
<div id="overview-2" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Overview</h2>
<p>Now that you are familiar with <a href="summarizedexperiment.html#summarizedexperiment"><code>SummarizedExperiment</code></a> and <a href="design-matrices.html#design-matrices">design matrices</a>, you can apply what you learned to the <code>airway</code> dataset.</p>
<p>Recall that the <code>airway</code> data is from an RNA-Seq experiment on four human airway smooth muscle cell lines treated with dexamethasone. You can learn more about the experiment in <span class="citation">Himes (<a href="#ref-Himes2014" role="doc-biblioref">2014</a>)</span>.</p>
</div>
<div id="load-prepare-data" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Load &amp; Prepare Data</h2>
<p>Load the <code>airway</code> dataset. You might need to install it.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="simple-model-fitting.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AnVIL::install(&quot;airway&quot;)</span></span>
<span id="cb78-2"><a href="simple-model-fitting.html#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(airway)</span>
<span id="cb78-3"><a href="simple-model-fitting.html#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(airway)</span></code></pre></div>
<p>You should also save the assay data from <code>airway</code>. Remember that <code>assay()</code> extracts the gene counts.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="simple-model-fitting.html#cb79-1" aria-hidden="true" tabindex="-1"></a>assay_data <span class="ot">&lt;-</span> <span class="fu">assay</span>(airway)</span></code></pre></div>
</div>
<div id="creating-the-design-matrix" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Creating the Design Matrix</h2>
<p>You should create a design matrix like you did in the <a href="design-matrices.html#inpractice">previous chapter</a>.</p>
<p>Since the dexamethasone treatment is a categorical variable, you need to use a <a href="design-matrices.html#means">means</a> or <a href="design-matrices.html#meanref">mean-reference</a> model. Going forward, we will use a mean-reference model because we are interested in the difference attributable to the treatment.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="simple-model-fitting.html#cb80-1" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">colData</span>(airway)</span>
<span id="cb80-2"><a href="simple-model-fitting.html#cb80-2" aria-hidden="true" tabindex="-1"></a>design_matrix <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> sample_data<span class="sc">$</span>dex)</span></code></pre></div>
</div>
<div id="testing-with-lmfit-and-ebayes" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Testing with <code>lmFit</code> and <code>eBayes</code></h2>
<p>Now you start the exciting process of model fitting! You will need the <code>limma</code> package.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="simple-model-fitting.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AnVIL::install(&quot;limma&quot;)</span></span>
<span id="cb81-2"><a href="simple-model-fitting.html#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span></code></pre></div>
<p>You want to transform the count data before proceeding with model fitting.</p>
<p>The <code>voom()</code> function transforms count data to <span class="math inline">\(\log_{2}\)</span>-counts per million (logCPM), estimates the mean-variance relationship, and uses these values to compute appropriate observation-level weights. This is an important step that takes into account the mean-variance relationship in the data. In other words, it factors in that as gene counts (expression) becomes greater, so does the variance. <code>voom()</code> also corrects for differences in sequencing depth among samples (libraries). This can happen when one sample starts with more total RNA or the sequencer produces more data for one sample over another.</p>
<p>You should also supply the design matrix.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="simple-model-fitting.html#cb82-1" aria-hidden="true" tabindex="-1"></a>assay_data <span class="ot">&lt;-</span> <span class="fu">voom</span>(assay_data, <span class="at">design =</span> design_matrix)</span></code></pre></div>
<p>Next, use the <code>limma</code> function <code>lmFit()</code>. This function fits multiple linear models by weighted or generalized least squares. A linear model is fitted to the expression data for each gene. As above, you should supply the design matrix.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="simple-model-fitting.html#cb83-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(assay_data, <span class="at">design =</span> design_matrix)</span></code></pre></div>
<p>The code above estimates coefficients, but it’s easier to interpret if you have log-odd or p-values. You can use <code>eBayes()</code> for this.</p>
<p>The <code>eBayes()</code> function computes moderated t-statistics, moderated F-statistic, and log-odds of differential expression by empirical Bayes moderation of the standard errors, when this function is given the output of <code>lmFit()</code>. This allows us to rank genes in order of evidence for differential expression. Using empirical Bayes methods squeezes the gene-wise residual variances toward a common value (or towards a global trend).</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="simple-model-fitting.html#cb84-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit)</span></code></pre></div>
<p>Now that you have a fitted model and estimated statistics, you can look at individual genes.</p>
<p>Because the fitted models are based on the design matrix we supplied, “top” genes will be based the difference between <code>trt</code> and <code>untrt</code>. Remember that these indicate whether or not cells received dexamethasone treatment. The function <code>topTable()</code> shows you the top-ranked genes given the model you supplied.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="simple-model-fitting.html#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit)</span></code></pre></div>
<pre><code>## Removing intercept from test coefficients</code></pre>
<pre><code>##                     logFC  AveExpr         t      P.Value    adj.P.Val
## ENSG00000134686 -1.304196 6.837469 -22.28727 1.384466e-09 8.874707e-05
## ENSG00000148175 -1.364975 8.854588 -18.96349 6.315648e-09 1.428428e-04
## ENSG00000162614 -1.945885 7.636408 -17.51779 1.325278e-08 1.428428e-04
## ENSG00000162616 -1.430717 5.903480 -17.27478 1.509630e-08 1.428428e-04
## ENSG00000189221 -3.263290 5.949117 -16.91641 1.835153e-08 1.428428e-04
## ENSG00000152583 -4.480843 4.165462 -17.88614 1.091355e-08 1.428428e-04
## ENSG00000120129 -2.847874 6.643013 -16.56703 2.228368e-08 1.428428e-04
## ENSG00000178695  2.589942 6.450283  16.57994 2.212289e-08 1.428428e-04
## ENSG00000196517  2.289351 3.024889  17.85198 1.111011e-08 1.428428e-04
## ENSG00000125148 -2.136635 7.022630 -15.98043 3.114408e-08 1.814907e-04
##                         B
## ENSG00000134686 12.547426
## ENSG00000148175 11.263744
## ENSG00000162614 10.516600
## ENSG00000162616 10.307478
## ENSG00000189221 10.103458
## ENSG00000152583 10.049829
## ENSG00000120129  9.977213
## ENSG00000178695  9.974713
## ENSG00000196517  9.831402
## ENSG00000125148  9.673848</code></pre>
</div>
<div id="interpreting-toptable-output" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Interpreting <code>topTable()</code> Output</h2>
<p>You now have the top genes affected by the treatment. Here’s how you can interpret the output.</p>
<ul>
<li><p><strong>logFC</strong>: estimate of the log2-fold-change corresponding to the effect of the treatment. Negative values mean the second treatment, in this case <code>untrt</code>, has lower expression than the baseline, in this case <code>trt</code>. Take a look at the design matrix to confirm.</p></li>
<li><p><strong>AveExpr </strong>: average log2-expression for the gene over all samples.</p></li>
<li><p><strong>t</strong>: moderated t-statistic</p></li>
<li><p><strong>P.Value</strong>: raw p-value</p></li>
<li><p><strong>adj.P.Value</strong>: adjusted p-value using the Benjamini &amp; Hochberg method (by default)</p></li>
<li><p><strong>B</strong>: log-odds that the gene is differentially expressed</p></li>
</ul>
<div class="reflection">
<p>QUESTIONS:</p>
<ol style="list-style-type: decimal">
<li>The top three genes have a negative logFC. Which treatment (<code>untrt</code> or <code>trt</code>) had greater expression?</li>
<li><code>eBayes()</code> performs p-value adjustment automatically. Why is it essential that we perform p-value adjustment for gene expression data?</li>
</ol>
</div>
</div>
<div id="recap-3" class="section level2" number="8.6">
<h2><span class="header-section-number">8.6</span> Recap</h2>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="simple-model-fitting.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.3 (2022-03-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] limma_3.50.3                airway_1.14.0              
##  [3] SummarizedExperiment_1.24.0 Biobase_2.54.0             
##  [5] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
##  [7] IRanges_2.28.0              S4Vectors_0.32.4           
##  [9] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
## [11] matrixStats_0.61.0         
## 
## loaded via a namespace (and not attached):
##  [1] bslib_0.3.1            compiler_4.1.3         jquerylib_0.1.4       
##  [4] XVector_0.34.0         bitops_1.0-7           tools_4.1.3           
##  [7] zlibbioc_1.40.0        digest_0.6.29          jsonlite_1.8.0        
## [10] evaluate_0.15          lattice_0.20-45        rlang_1.0.2           
## [13] Matrix_1.4-0           DelayedArray_0.20.0    cli_3.2.0             
## [16] yaml_2.3.5             xfun_0.26              fastmap_1.1.0         
## [19] GenomeInfoDbData_1.2.7 stringr_1.4.0          knitr_1.33            
## [22] sass_0.4.1             grid_4.1.3             R6_2.5.1              
## [25] rmarkdown_2.10         bookdown_0.24          magrittr_2.0.3        
## [28] htmltools_0.5.2        stringi_1.7.6          RCurl_1.98-1.6</code></pre>

</div>
</div>



<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Himes2014" class="csl-entry">
Himes, Xiaofeng AND Wagner, Blanca E. AND Jiang. 2014. <span>“RNA-Seq Transcriptome Profiling Identifies Crispld2 as a Glucocorticoid Responsive Gene That Modulates Cytokine Function in Airway Smooth Muscle Cells.”</span> <em>PLOS ONE</em> 9 (6): 1–13. <a href="https://doi.org/10.1371/journal.pone.0099625">https://doi.org/10.1371/journal.pone.0099625</a>.
</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="design-matrices.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="help.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
