<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Transformation, Fold-change, &amp; MA Plots | Statistics for Genomics: Differential Expression</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Transformation, Fold-change, &amp; MA Plots | Statistics for Genomics: Differential Expression" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Transformation, Fold-change, &amp; MA Plots | Statistics for Genomics: Differential Expression" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2022-08-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/anvil_favicon.ico" type="image/x-icon" />
<link rel="prev" href="summarizedexperiment.html"/>
<link rel="next" href="design-matrices.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style_anvil.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://anvilproject.org/" target="_blank"><img src="assets/logo-anvil.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-collection"><i class="fa fa-check"></i>AnVIL Collection</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html"><i class="fa fa-check"></i><b>1</b> Overview and Learning Objectives</a>
<ul>
<li class="chapter" data-level="1.1" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html#activity-context"><i class="fa fa-check"></i><b>1.1</b> Activity Context</a></li>
<li class="chapter" data-level="1.2" data-path="overview-and-learning-objectives.html"><a href="overview-and-learning-objectives.html#learning-objectives"><i class="fa fa-check"></i><b>1.2</b> Learning Objectives</a></li>
</ul></li>
<li class="part"><span><b>RStudio on AnVIL</b></span></li>
<li class="chapter" data-level="2" data-path="getting-set-up.html"><a href="getting-set-up.html"><i class="fa fa-check"></i><b>2</b> Getting Set Up</a>
<ul>
<li class="chapter" data-level="2.1" data-path="getting-set-up.html"><a href="getting-set-up.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="getting-set-up.html"><a href="getting-set-up.html#create-a-google-account"><i class="fa fa-check"></i><b>2.2</b> Create a Google Account</a></li>
<li class="chapter" data-level="2.3" data-path="getting-set-up.html"><a href="getting-set-up.html#sign-in-to-terra"><i class="fa fa-check"></i><b>2.3</b> Sign in to Terra</a></li>
<li class="chapter" data-level="2.4" data-path="getting-set-up.html"><a href="getting-set-up.html#share-your-id"><i class="fa fa-check"></i><b>2.4</b> Share your ID</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="workspace.html"><a href="workspace.html"><i class="fa fa-check"></i><b>3</b> Creating a Workspace</a></li>
<li class="chapter" data-level="4" data-path="rstudio.html"><a href="rstudio.html"><i class="fa fa-check"></i><b>4</b> Working in RStudio</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rstudio.html"><a href="rstudio.html#launch-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.1</b> Launch RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.2" data-path="rstudio.html"><a href="rstudio.html#tour-rstudio"><i class="fa fa-check"></i><b>4.2</b> Tour RStudio</a></li>
<li class="chapter" data-level="4.3" data-path="rstudio.html"><a href="rstudio.html#more-practice-with-isee"><i class="fa fa-check"></i><b>4.3</b> More Practice with <code>iSEE</code></a></li>
<li class="chapter" data-level="4.4" data-path="rstudio.html"><a href="rstudio.html#stopping"><i class="fa fa-check"></i><b>4.4</b> Pause RStudio</a></li>
<li class="chapter" data-level="4.5" data-path="rstudio.html"><a href="rstudio.html#delete-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.5</b> Delete RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.6" data-path="rstudio.html"><a href="rstudio.html#video-guide"><i class="fa fa-check"></i><b>4.6</b> Video Guide</a></li>
</ul></li>
<li class="part"><span><b>EXPRESSION DATA</b></span></li>
<li class="chapter" data-level="5" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html"><i class="fa fa-check"></i><b>5</b> The <code>SummarizedExperiment</code> class</a>
<ul>
<li class="chapter" data-level="5.1" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html#exploring-summarizedexperiment"><i class="fa fa-check"></i><b>5.2</b> Exploring <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="5.3" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html#subsetting-summarizedexperiment"><i class="fa fa-check"></i><b>5.3</b> Subsetting <code>SummarizedExperiment</code></a></li>
<li class="chapter" data-level="5.4" data-path="summarizedexperiment.html"><a href="summarizedexperiment.html#recap"><i class="fa fa-check"></i><b>5.4</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html"><i class="fa fa-check"></i><b>6</b> Transformation, Fold-change, &amp; MA Plots</a>
<ul>
<li class="chapter" data-level="6.1" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#gene-expression-data"><i class="fa fa-check"></i><b>6.1</b> Gene Expression Data</a></li>
<li class="chapter" data-level="6.2" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#plotting-raw-counts"><i class="fa fa-check"></i><b>6.2</b> Plotting Raw Counts</a></li>
<li class="chapter" data-level="6.3" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#transforming-log_2"><i class="fa fa-check"></i><b>6.3</b> Transforming (<span class="math inline">\(\log_{2}\)</span>)</a></li>
<li class="chapter" data-level="6.4" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#using-fold-change-to-create-an-ma-plot"><i class="fa fa-check"></i><b>6.4</b> Using Fold-Change to Create an MA Plot</a></li>
<li class="chapter" data-level="6.5" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#future-directions"><i class="fa fa-check"></i><b>6.5</b> Future Directions</a></li>
<li class="chapter" data-level="6.6" data-path="transformation-fold-change-ma-plots.html"><a href="transformation-fold-change-ma-plots.html#recap-1"><i class="fa fa-check"></i><b>6.6</b> Recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="design-matrices.html"><a href="design-matrices.html"><i class="fa fa-check"></i><b>7</b> Design Matrices</a>
<ul>
<li class="chapter" data-level="7.1" data-path="design-matrices.html"><a href="design-matrices.html#overview-1"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="design-matrices.html"><a href="design-matrices.html#types-of-models"><i class="fa fa-check"></i><b>7.2</b> Types of Models</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="design-matrices.html"><a href="design-matrices.html#regression-continuous-variables"><i class="fa fa-check"></i><b>7.2.1</b> Regression: Continuous Variables</a></li>
<li class="chapter" data-level="7.2.2" data-path="design-matrices.html"><a href="design-matrices.html#means-categorical-variables"><i class="fa fa-check"></i><b>7.2.2</b> Means: Categorical Variables</a></li>
<li class="chapter" data-level="7.2.3" data-path="design-matrices.html"><a href="design-matrices.html#mean-reference-categorical-variables"><i class="fa fa-check"></i><b>7.2.3</b> Mean-Reference: Categorical Variables</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="design-matrices.html"><a href="design-matrices.html#making-a-design-matrix"><i class="fa fa-check"></i><b>7.3</b> Making a Design Matrix</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="design-matrices.html"><a href="design-matrices.html#regression-continuous-variables-1"><i class="fa fa-check"></i><b>7.3.1</b> Regression: Continuous Variables</a></li>
<li class="chapter" data-level="7.3.2" data-path="design-matrices.html"><a href="design-matrices.html#means"><i class="fa fa-check"></i><b>7.3.2</b> Means: Categorical Variables</a></li>
<li class="chapter" data-level="7.3.3" data-path="design-matrices.html"><a href="design-matrices.html#meanref"><i class="fa fa-check"></i><b>7.3.3</b> Mean-Reference: Categorical Variables</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="design-matrices.html"><a href="design-matrices.html#more-complex-designs"><i class="fa fa-check"></i><b>7.4</b> More Complex Designs</a></li>
<li class="chapter" data-level="7.5" data-path="design-matrices.html"><a href="design-matrices.html#inpractice"><i class="fa fa-check"></i><b>7.5</b> Design Matrix in Practice</a></li>
<li class="chapter" data-level="7.6" data-path="design-matrices.html"><a href="design-matrices.html#recap-2"><i class="fa fa-check"></i><b>7.6</b> Recap</a></li>
</ul></li>
<li class="part"><span><b>PUTTING IT TOGETHER</b></span></li>
<li class="chapter" data-level="8" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html"><i class="fa fa-check"></i><b>8</b> Simple Model Fitting</a>
<ul>
<li class="chapter" data-level="8.1" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#overview-2"><i class="fa fa-check"></i><b>8.1</b> Overview</a></li>
<li class="chapter" data-level="8.2" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#load-prepare-data"><i class="fa fa-check"></i><b>8.2</b> Load &amp; Prepare Data</a></li>
<li class="chapter" data-level="8.3" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#creating-the-design-matrix"><i class="fa fa-check"></i><b>8.3</b> Creating the Design Matrix</a></li>
<li class="chapter" data-level="8.4" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#testing-with-lmfit-and-ebayes"><i class="fa fa-check"></i><b>8.4</b> Testing with <code>lmFit</code> and <code>eBayes</code></a></li>
<li class="chapter" data-level="8.5" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#interpreting-toptable-output"><i class="fa fa-check"></i><b>8.5</b> Interpreting <code>topTable()</code> Output</a></li>
<li class="chapter" data-level="8.6" data-path="simple-model-fitting.html"><a href="simple-model-fitting.html#recap-3"><i class="fa fa-check"></i><b>8.6</b> Recap</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="help.html"><a href="help.html"><i class="fa fa-check"></i><b>A</b> Help</a></li>
<li class="chapter" data-level="B" data-path="give-us-feedback.html"><a href="give-us-feedback.html"><i class="fa fa-check"></i><b>B</b> Give Us Feedback</a></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/DaSL_Course_Template_Bookdown" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistics for Genomics: Differential Expression</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
  <link href="assets/style_anvil.css" rel="stylesheet">
</head>
        


<div class="hero-image-container"> 
  <img class= "hero-image" src="assets/dasl_thin_main_image.png">
</div>
<div id="transformation-fold-change-ma-plots" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> Transformation, Fold-change, &amp; MA Plots</h1>
<p>The following excerpt comes from <span class="citation">Ritchie et al. (<a href="#ref-Ritchie2015" role="doc-biblioref">2015</a>)</span> :</p>
<blockquote>
<p><em>Measuring expression in multiple RNA samples produces columns of correlated expression values, which are highly correlated because they are measured on the same set of genes or genomic features. It has long been established in the biomedical literature that the level of agreement between correlated variables can be usefully examined by plotting differences versus means.</em></p>
</blockquote>
<p>In other words, gene expression data is full of correlations. We must think carefully about how we examine and plot gene expression data. In the next steps, we will:</p>
<ul>
<li>Examine how <span class="math inline">\(\log_{2}\)</span> transformations and fold-change improve data clarity</li>
<li>Learn how to make MA plots on gene expression data</li>
</ul>
<p>Much of the following has been adopted from the <a href="https://bioconductor.org/packages/release/bioc/vignettes/Glimma/inst/doc/limma_edger.html"><code>Glimma</code> vignette</a> for <code>limma</code> and <code>edgeR</code> <span class="citation">(<a href="#ref-Su2017" role="doc-biblioref">Su et al. 2017</a>)</span>.</p>
<div id="gene-expression-data" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Gene Expression Data</h2>
<p>First, we will load the necessary packages.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="transformation-fold-change-ma-plots.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Install and load airway</span></span>
<span id="cb28-2"><a href="transformation-fold-change-ma-plots.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co"># AnVIL::install(c(&quot;airway&quot;))</span></span>
<span id="cb28-3"><a href="transformation-fold-change-ma-plots.html#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(airway)</span></code></pre></div>
<p>Load the gene expression data. We will be using data from an RNA-Seq experiment on four human airway smooth muscle cell lines treated with dexamethasone <span class="citation">(<a href="#ref-Himes2014" role="doc-biblioref">Himes 2014</a>)</span>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="transformation-fold-change-ma-plots.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the gene expression data</span></span>
<span id="cb29-2"><a href="transformation-fold-change-ma-plots.html#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(airway)</span>
<span id="cb29-3"><a href="transformation-fold-change-ma-plots.html#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">assay</span>(airway),<span class="dv">10</span>)</span></code></pre></div>
<pre><code>##                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
## ENSG00000000003        679        448        873        408       1138
## ENSG00000000005          0          0          0          0          0
## ENSG00000000419        467        515        621        365        587
## ENSG00000000457        260        211        263        164        245
## ENSG00000000460         60         55         40         35         78
## ENSG00000000938          0          0          2          0          1
## ENSG00000000971       3251       3679       6177       4252       6721
## ENSG00000001036       1433       1062       1733        881       1424
## ENSG00000001084        519        380        595        493        820
## ENSG00000001167        394        236        464        175        658
##                 SRR1039517 SRR1039520 SRR1039521
## ENSG00000000003       1047        770        572
## ENSG00000000005          0          0          0
## ENSG00000000419        799        417        508
## ENSG00000000457        331        233        229
## ENSG00000000460         63         76         60
## ENSG00000000938          0          0          0
## ENSG00000000971      11027       5176       7995
## ENSG00000001036       1439       1359       1109
## ENSG00000001084        714        696        704
## ENSG00000001167        584        360        269</code></pre>
<p>In this data, each sample is in a column, while each gene is in a row. These raw counts range from very small to quite large numbers!</p>
</div>
<div id="plotting-raw-counts" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Plotting Raw Counts</h2>
<p>Let’s say we want to compare treated and untreated samples. We will first look at the sample data.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="transformation-fold-change-ma-plots.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># View metadata</span></span>
<span id="cb31-2"><a href="transformation-fold-change-ma-plots.html#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(airway)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 9 columns
##            SampleName     cell      dex    albut        Run avgLength
##              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;
## SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126
## SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126
## SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126
## SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87
## SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120
## SRR1039517 GSM1275871  N080611    trt      untrt SRR1039517       126
## SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101
## SRR1039521 GSM1275875  N061011    trt      untrt SRR1039521        98
##            Experiment    Sample    BioSample
##              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;
## SRR1039508  SRX384345 SRS508568 SAMN02422669
## SRR1039509  SRX384346 SRS508567 SAMN02422675
## SRR1039512  SRX384349 SRS508571 SAMN02422678
## SRR1039513  SRX384350 SRS508572 SAMN02422670
## SRR1039516  SRX384353 SRS508575 SAMN02422682
## SRR1039517  SRX384354 SRS508576 SAMN02422673
## SRR1039520  SRX384357 SRS508579 SAMN02422683
## SRR1039521  SRX384358 SRS508580 SAMN02422677</code></pre>
<p>We will manually take the mean across treated and untreated samples.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="transformation-fold-change-ma-plots.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Collect the counts and sample data</span></span>
<span id="cb33-2"><a href="transformation-fold-change-ma-plots.html#cb33-2" aria-hidden="true" tabindex="-1"></a>raw_counts <span class="ot">&lt;-</span> <span class="fu">assay</span>(airway)</span>
<span id="cb33-3"><a href="transformation-fold-change-ma-plots.html#cb33-3" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">colData</span>(airway)</span>
<span id="cb33-4"><a href="transformation-fold-change-ma-plots.html#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="transformation-fold-change-ma-plots.html#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Select untreated samples</span></span>
<span id="cb33-6"><a href="transformation-fold-change-ma-plots.html#cb33-6" aria-hidden="true" tabindex="-1"></a>untrt_sample_means <span class="ot">&lt;-</span> <span class="fu">rowMeans2</span>(raw_counts[, sample_data<span class="sc">$</span>dex <span class="sc">==</span> <span class="st">&quot;untrt&quot;</span>])</span>
<span id="cb33-7"><a href="transformation-fold-change-ma-plots.html#cb33-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-8"><a href="transformation-fold-change-ma-plots.html#cb33-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Select treated samples</span></span>
<span id="cb33-9"><a href="transformation-fold-change-ma-plots.html#cb33-9" aria-hidden="true" tabindex="-1"></a>trt_sample_means <span class="ot">&lt;-</span> <span class="fu">rowMeans2</span>(raw_counts[, sample_data<span class="sc">$</span>dex <span class="sc">==</span> <span class="st">&quot;trt&quot;</span>])</span></code></pre></div>
<p>Then we will plot the two groups.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="transformation-fold-change-ma-plots.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(untrt_sample_means, trt_sample_means)</span></code></pre></div>
<p><img src="06-transform-MAplots_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>This is hard to interpret. Most of the genes are clustered in the bottom left corner.</p>
</div>
<div id="transforming-log_2" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Transforming (<span class="math inline">\(\log_{2}\)</span>)</h2>
<p>Using a <span class="math inline">\(\log_{2}\)</span> transformation makes it easier to examine all the genes together.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="transformation-fold-change-ma-plots.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot with log2 transformation</span></span>
<span id="cb35-2"><a href="transformation-fold-change-ma-plots.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">log2</span>(untrt_sample_means), <span class="fu">log2</span>(trt_sample_means))</span></code></pre></div>
<p><img src="06-transform-MAplots_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="using-fold-change-to-create-an-ma-plot" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Using Fold-Change to Create an MA Plot</h2>
<p>As stated at the very start of this chapter, plotting differences versus means can be very helpful when many genes are correlated. It also makes interpretation easier when combined with a <span class="math inline">\(\log_{2}\)</span> transformation.</p>
<p>In the next steps, you will create a preliminary MA plot. MA plots are a widely used way to visualize genomic data. The <em><strong>M</strong></em> represents the difference between two conditions (fold-change), while the <em><strong>A</strong></em> represents the average intensity of the expression. Both values take on a <span class="math inline">\(\log_{2}\)</span> transformation.</p>
<p><em><strong>M</strong></em> is expressed as a log ratio or difference in the following form. <em><strong>M</strong></em> is almost always placed on the y-axis.</p>
<p><span class="math display">\[M = log_{2}(\frac{condition 1}{condition 2}) = log_{2}(condition 1) - log_{2}(condition2)\]</span></p>
<p><em><strong>A</strong></em> is more simple, taking the form of a transformed average. <em><strong>A</strong></em> is often called “log mean expression” and is almost always placed on the x-axis.</p>
<p><span class="math display">\[A = \frac{1}{2} (log_{2}(condition 1) + log_{2}(condition 2))\]</span></p>
<p>When untreated and treated expression are equal, fold-change is equal to zero.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="transformation-fold-change-ma-plots.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">log2</span>(<span class="dv">10</span>) <span class="sc">-</span> <span class="fu">log2</span>(<span class="dv">10</span>)</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>When untreated expression is greater, fold-change is positive.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="transformation-fold-change-ma-plots.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">log2</span>(<span class="dv">20</span>) <span class="sc">-</span> <span class="fu">log2</span>(<span class="dv">10</span>)</span></code></pre></div>
<pre><code>## [1] 1</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="transformation-fold-change-ma-plots.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">log2</span>(<span class="dv">100</span>) <span class="sc">-</span> <span class="fu">log2</span>(<span class="dv">1</span>)</span></code></pre></div>
<pre><code>## [1] 6.643856</code></pre>
<p>However when treated expression is greater, fold-change is negative.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="transformation-fold-change-ma-plots.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">log2</span>(<span class="dv">10</span>) <span class="sc">-</span> <span class="fu">log2</span>(<span class="dv">20</span>)</span></code></pre></div>
<pre><code>## [1] -1</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="transformation-fold-change-ma-plots.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">log2</span>(<span class="dv">1</span>) <span class="sc">-</span> <span class="fu">log2</span>(<span class="dv">100</span>)</span></code></pre></div>
<pre><code>## [1] -6.643856</code></pre>
<p>Calculate log2 fold-change and mean expression for the data.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="transformation-fold-change-ma-plots.html#cb46-1" aria-hidden="true" tabindex="-1"></a>log2_fold_change <span class="ot">&lt;-</span> <span class="fu">log2</span>(untrt_sample_means) <span class="sc">-</span> <span class="fu">log2</span>(trt_sample_means)</span>
<span id="cb46-2"><a href="transformation-fold-change-ma-plots.html#cb46-2" aria-hidden="true" tabindex="-1"></a>mean_expression <span class="ot">&lt;-</span> (<span class="fu">log2</span>(untrt_sample_means) <span class="sc">+</span> <span class="fu">log2</span>(trt_sample_means)) <span class="sc">/</span> <span class="dv">2</span></span></code></pre></div>
<p>Plot the values above, with mean expression on the x-axis and fold-change on the y-axis.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="transformation-fold-change-ma-plots.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mean_expression, log2_fold_change)</span></code></pre></div>
<p><img src="06-transform-MAplots_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>Since there are many data points, it’s a good idea to customize your plot. The <code>ggplot2</code> package can also provide greater flexibility with plotting.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="transformation-fold-change-ma-plots.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb48-2"><a href="transformation-fold-change-ma-plots.html#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mean_expression, log2_fold_change, <span class="at">col =</span> <span class="fu">alpha</span>(<span class="st">&quot;black&quot;</span>, <span class="fl">0.1</span>), <span class="at">pch =</span> <span class="dv">16</span>, <span class="at">cex =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="06-transform-MAplots_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>This gives us a very rough idea of how transformation and using fold-change can aid in interpretation of the data. In reality, we need to cover a few more steps before creating this kind of plot.</p>
</div>
<div id="future-directions" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Future Directions</h2>
<p>A simple MA plot can be produced with the <code>plotMA()</code> function from the <code>limma</code> package. Other packages for exploring differential gene expression in R, such as <code>DESeq2</code>, also have functions to create MA plots.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="transformation-fold-change-ma-plots.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Not run</span></span>
<span id="cb49-2"><a href="transformation-fold-change-ma-plots.html#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>()</span></code></pre></div>
<p><code>glimmaMA()</code> from the <code>Glimma</code> package creates interactive MA plots.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="transformation-fold-change-ma-plots.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Not run</span></span>
<span id="cb50-2"><a href="transformation-fold-change-ma-plots.html#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">glimmaMA</span>()</span></code></pre></div>
</div>
<div id="recap-1" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> Recap</h2>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="transformation-fold-change-ma-plots.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.3 (2022-03-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] scales_1.2.0                airway_1.14.0              
##  [3] SummarizedExperiment_1.24.0 Biobase_2.54.0             
##  [5] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
##  [7] IRanges_2.28.0              S4Vectors_0.32.4           
##  [9] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
## [11] matrixStats_0.61.0         
## 
## loaded via a namespace (and not attached):
##  [1] highr_0.9              bslib_0.3.1            compiler_4.1.3        
##  [4] jquerylib_0.1.4        XVector_0.34.0         bitops_1.0-7          
##  [7] tools_4.1.3            zlibbioc_1.40.0        digest_0.6.29         
## [10] lifecycle_1.0.1        jsonlite_1.8.0         evaluate_0.15         
## [13] lattice_0.20-45        rlang_1.0.2            Matrix_1.4-0          
## [16] DelayedArray_0.20.0    cli_3.2.0              yaml_2.3.5            
## [19] xfun_0.26              fastmap_1.1.0          GenomeInfoDbData_1.2.7
## [22] stringr_1.4.0          knitr_1.33             sass_0.4.1            
## [25] grid_4.1.3             R6_2.5.1               rmarkdown_2.10        
## [28] bookdown_0.24          farver_2.1.0           magrittr_2.0.3        
## [31] htmltools_0.5.2        colorspace_2.0-3       stringi_1.7.6         
## [34] munsell_0.5.0          RCurl_1.98-1.6</code></pre>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Himes2014" class="csl-entry">
Himes, Xiaofeng AND Wagner, Blanca E. AND Jiang. 2014. <span>“RNA-Seq Transcriptome Profiling Identifies Crispld2 as a Glucocorticoid Responsive Gene That Modulates Cytokine Function in Airway Smooth Muscle Cells.”</span> <em>PLOS ONE</em> 9 (6): 1–13. <a href="https://doi.org/10.1371/journal.pone.0099625">https://doi.org/10.1371/journal.pone.0099625</a>.
</div>
<div id="ref-Ritchie2015" class="csl-entry">
Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law, Wei Shi, and Gordon K. Smyth. 2015. <span>“<span class="nocase">limma powers differential expression analyses for RNA-sequencing and microarray studies</span>.”</span> <em>Nucleic Acids Research</em> 43 (7): e47–47. <a href="https://doi.org/10.1093/nar/gkv007">https://doi.org/10.1093/nar/gkv007</a>.
</div>
<div id="ref-Su2017" class="csl-entry">
Su, Shian, Charity W Law, Casey Ah-Cann, Marie-Liesse Asselin-Labat, Marnie E Blewitt, and Matthew E Ritchie. 2017. <span>“<span class="nocase">Glimma: interactive graphics for gene expression analysis</span>.”</span> <em>Bioinformatics</em> 33 (13): 2050–52. <a href="https://doi.org/10.1093/bioinformatics/btx094">https://doi.org/10.1093/bioinformatics/btx094</a>.
</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="summarizedexperiment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="design-matrices.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/jhudsl/GDSCN_Book_Statistics_for_Genomics_Differential_Expression/edit/main/06-transform-MAplots.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
